### DISCUSSION TITLE
How to manage build scripts in monorepo?

### DISCUSSION THREAD

**Discussion by gweesin** [pnpm/pnpm#7142] [Q&A] [answered]
My workspace is organized as follows:

```txt
packages:
  - moduleA
  - moduleB
  - ...
  - moduleZ
```

All these modules share the same build steps and use common configuration files like `check-legal.js`, `webpack.dev.js`, and so on. It's worth noting that `check-legal.js` performs some checks before publishing.

Now, I want to extract this file and place it in another module called `@scripts/check-legal`. In the `package.json` of moduleA, I have added the following script:

```json
{
  "scripts": {
    "check-legal": "pnpm run --filter @scripts/check-legal -- check $(pwd)"
  }
}
```

However, I'm facing an issue where $(pwd) is treated as a constant string. Do you have any suggestions to address this problem?"

**Comment by gweesin** [comment]
This question might be related to issue #3166, but I haven't found a solution to this problem yet.

**Reply by zkochan** [reply, marked as answer]
In the pnpm monorepo we use [this tool](https://github.com/pnpm/meta-updater) to sync scripts in all projects

**Reply by gweesin** [reply]
LGTM!
